src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java

```diff
@@ -266,45 +322,78 @@ public Response createSession(fi.metatavu.metamind.rest.model.Session body, UUID
       return createBadRequest("Invalid story parameter");
     }
     
-    UUID loggedUserId = getLoggerUserId();
     String locale = body.getLocale();
     String visitor = body.getVisitor();
     
-    // TODO: Permission check
-    
     Session session = sessionController.create(story, locale, body.getTimeZone(), visitor, loggedUserId);
     
+    try {
+      UUID createdResourceId = authenticationController.createProtectedResource(loggedUserId, String.format("Session-%s", session.getId()), String.format("/v2/stories/sessions/%s", session.getId()), "session", SCOPES);
```

Comment:
Sessions define only, well sessions, so they do not require authorization resources